pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -285,19 +285,7 @@ func (a *index) Lookup(key string) []model.AddressInfo {
 
 	// 2. Workload by IP
 	if wls := a.workloads.ByAddress.Lookup(networkAddr); len(wls) > 0 {
-		// If there is just one, return it
-		if len(wls) == 1 {
-			return []model.AddressInfo{modelWorkloadToAddressInfo(wls[0])}
-		}
-		// Otherwise, try to find a pod - pods have precedence
-		pod := slices.FindFunc(wls, func(info model.WorkloadInfo) bool {
-			return info.Source == kind.Pod
-		})
-		if pod != nil {
-			return []model.AddressInfo{modelWorkloadToAddressInfo(*pod)}
-		}
-		// Otherwise just return the first one; all WorkloadEntry have the same weight
-		return []model.AddressInfo{modelWorkloadToAddressInfo(wls[0])}
+		return dedupeWorkloads(wls)
```

Comment:
Refactor to share code with Lookup() and All(); the actual logic changes too.
* Lookup() can return multiple now
* Only exclude dupes for !hostnetwork